### DISCUSSION TITLE
not able to convert this smile to mol.

### DISCUSSION THREAD

**Discussion by sharmaphys** [rdkit/rdkit#5881] [Q&A]
While I am able to convert all the smiles in a commercial database, it gives me the below error trying to convert smile ("Br[Br]Br.c1ccncc1") to mol.

mol = Chem.MolFromSmiles("Br[Br]Br.c1ccncc1")
Could anyone suggest how to solve the issue?

=============================================================
ArgumentError                             Traceback (most recent call last)
ArgumentError: Python argument types in
    FilterCatalog.GetFirstMatch(FilterCatalog, NoneType)
did not match C++ signature:
    GetFirstMatch(RDKit::FilterCatalog {lvalue}, RDKit::ROMol mol)

**Comment by manajit-das** [comment]
Try this: 'Br[Br+]Br.c1ccncc1'
In the given SMILES the valence of the middle Bromine atom is not right, hence the problem I think.
![smiles](https://user-images.githubusercontent.com/59758360/208226811-60ec86a7-d244-4348-81fd-9a10bbd1a147.PNG)

**Reply by sharmaphys** [reply]
@manajit-das 
The compound looks like spitted. It is supposed to be a single molecule. It does not look right to me. Please correct me if I am wrong.

**Reply by manajit-das** [reply]
It is not a single molecule. Two smiles string is concatenated by dot (.)
To my understanding,  it is not a single molecular entity. I think, RDKit considers valence of atoms carefully, if that is okay, then it will read the molecule, else it will show error.
Further, I believe in the database it actually corresponds to some reagent which is pyridinium tribromide. check the link here.
https://www.sigmaaldrich.com/IN/en/product/aldrich/133248?gclid=Cj0KCQiAtICdBhCLARIsALUBFcFi8Zf4Rk4NzC7LJNZOO6OBo2PzVktr0rOsCB0oxZu3YMYryd9RrSkaAlYiEALw_wcB&gclsrc=aw.ds
Hope it helps
